pkg/ccl/roleccl/role.go

```diff
@@ -95,6 +97,12 @@ func grantRolePlanHook(
 		return nil, nil, nil, false, nil
 	}
 
+	if grant.AdminOption {
```

Comment:
I think it would be cleaner for these functions to take in whether or not the command was with an AdminOption